packages/@o3r/testing/src/core/playwright/component-fixture.ts

```diff
@@ -50,15 +50,12 @@ export class O3rComponentFixture<V extends O3rElement = O3rElement> implements C
    * @param timeout specific timeout that will throw when reach
    */
   protected async throwOnUndefined<T extends O3rElement>(element: Promise<T | undefined>, timeout?: number): Promise<T> {
-    return withTimeout(element, timeout)
-      .then((el) => el?.sourceElement.element.count())
-      .then((count) => (count || 0) > 0)
-      .then((isPresent) => {
-        if (!isPresent) {
-          throw new Error('Element not found in ' + this.constructor.name);
-        }
-      })
-      .then(() => element as Promise<T>);
+    return withTimeout(
+      element.then(async (el) => {
+        await el?.sourceElement.element.first().waitFor({state: 'attached'});
+        return el!;
```

Comment:
But if `el == undefined` you will return `el!`.
It sounds like a lie :D

Reply:
In a playwright context when will you ever have el undefined if you do not pass it explicitely with a Promise.resolve(undefined)?
Locator always are defined, you create it from a query which has no chance to fail, the typing is only there for compatibility with the other frameworks.

The actual throwOnUndefined is handled line 55 where the timeout will throw if the element is not present in the page.

But yeah I will fix it to stay consistent with the signature

Reply:
As discussed, there is no impact in making the signature more restrictive and will go for this solution